packages/govern-contract-utils/test/adaptative-erc165.test.ts

```diff
@@ -0,0 +1,89 @@
+import { expect } from 'chai'
+import { Signer } from 'ethers'
+import { hexDataSlice, id } from 'ethers/lib/utils'
+import { ethers } from 'hardhat'
+import { AdaptativeErc165Mock, AdaptativeErc165MockFactory } from '../typechain'
+
+const ERRORS = {
+  UNKNOWN_CALLBACK: 'adap-erc165: unknown callback',
+}
+
+const EVENTS = {
+  REGISTERED_CALLBACK: 'RegisteredCallback',
+  REGISTERED_STANDARD: 'RegisteredStandard',
+  RECEIVED_CALLBACK: 'ReceivedCallback',
+}
+
+const beefInterfaceId = '0xbeefbeef'
+const callbackSig = hexDataSlice(id('callbackFunc()'), 0, 4) // 0x1eb2075a
+const magicNumber = '0x10000000'
+
+// NOTE: We cannot assert on the values emitted
+// in the events currently, see: https://github.com/EthWorks/Waffle/issues/87
+describe('AdaptativeErc165', function () {
+  let adaptative: AdaptativeErc165Mock, signers: Signer[]
+
+  before(async () => {
+    signers = await ethers.getSigners()
+    const Adaptative = (await ethers.getContractFactory(
+      'AdaptativeERC165Mock'
+    )) as AdaptativeErc165MockFactory
+    adaptative = await Adaptative.deploy()
+  })
+
+  context('registerStandardAndCallback', () => {
+    it('ensures the right events were fired', async () => {
+      await expect(
+        adaptative.registerStandardAndCallback(
+          beefInterfaceId,
+          callbackSig,
+          magicNumber
+        )
+      )
+        .to.emit(adaptative, EVENTS.REGISTERED_STANDARD)
+        .withArgs(beefInterfaceId)
+        .to.emit(adaptative, EVENTS.REGISTERED_CALLBACK)
+        .withArgs(callbackSig, magicNumber)
+    })
+
+    it('ensures support the right interfaceID', async () => {
+      await expect(
+        await adaptative.supportsInterface(beefInterfaceId)
+      ).to.equal(true)
+    })
+
+    it('ensures the right callback was call', async () => {
```

Comment:
We should here also test the inline assembly of the ``_handleCallback`` method. It does return the memory value at position ``0x00`` with an offset of ``0x20``.

Reply:
I'm not sure how to test this. How do you suggest we should do it?